### PR TITLE
Update ticketmatch for perforce.atlassian.net

### PR DESCRIPTION
**Pull request by joshcooper**
* Perforce JIRA does not allow anonymous access, so check for JIRA_AUTH_TOKEN when running `pa_matchbatch.sh` and provide instructions for how to set it.

```
$ unset JIRA_AUTH_TOKEN                                                                         
$ ./pa_matchbatch.sh main             
Error: The JIRA_AUTH_TOKEN environment variable must be set.

Create a token at https://id.atlassian.com/manage-profile/security/api-tokens and run

    export JIRA_AUTH_TOKEN=$(echo -n '<first>.<last>@perforce.com:<token>' | base64 --wrap=0)
```

* Use REST v3 API instead of v2

* Subtasks have "Done" state instead of "Resolved"

* Add links for problematic tickets to make it easier to bulk edit, for example to add "Not needed" to "Release Notes Summary", assign fixVersions, etc.

* Add links for all tickets fixed in that version, to make it easy to track what's fixed and correlated that with GH releases.

### CODE REVIEW

File: pa_matchbatch.sh [hunk 1]
```diff
@@ -22,7 +22,17 @@ overridesFile="/tmp/version_overrides.txt"
 OVERRIDE_PATH=$(realpath ${OVERRIDE_PATH:-${overridesFile}})
 
 # read in the auth token
-AUTH_TOKEN_ARG="-a ${JIRA_AUTH_TOKEN}"
+if [[ -z "$JIRA_AUTH_TOKEN" ]]; then
+    >&2 echo <<EOF
+Error: The JIRA_AUTH_TOKEN environment variable must be set.
+
+Create a token at https://id.atlassian.com/manage-profile/security/api-tokens and run
+export JIRA_API_TOKEN=$(echo -n '<username>@perforce.com:<token>' | base64 --wrap=0)
```
Comment (joshcooper) [inline pa_matchbatch.sh]: This isn't working as intended. Need to revisit

Comment (tvpartytonight) [inline pa_matchbatch.sh]: The `base64` that comes with macos doesn't have a `wrap` setting, but I believe the using `break` is equivalent.

Comment (joshcooper) [reply pa_matchbatch.sh]: Added a conditional based on `$OSTYPE`, though not sure it works as expected on macOS? Could you test?

Comment (tvpartytonight) [reply pa_matchbatch.sh]: I think it should be
```
 if [[ "$OSTYPE" == "darwin"* ]]; then
```

Changing it to that works for me.

Comment (joshcooper) [reply pa_matchbatch.sh]: @tvpartytonight I updated it

**Review by tvpartytonight — APPROVED** [review]

**Pull request merged**